src/realm.js

```diff
@@ -368,6 +371,23 @@ export class Realm {
     this.$GlobalEnv.environmentRecord.DeleteBinding(name);
   }
 
+  // Evaluate a context as if it won't have any side-effects outside of any objects
+  // that it created itself. This promises that any abstract functions inside of it
+  // also won't have effects on any objects or bindings that weren't created in this
+  // call.
+  evaluatePure<T>(f: () => T) {
+    let savedImpureObjects = this.impureObjects;
```

Comment:
Perhaps these objects should be termed "leaked objects" since they leak into the outside world and then effectively become abstract.

Reply:
They are not necessarily going to get returned from the pure function so they won't always leak that way. They also won't always leak into the called abstract function.

They only have the potential to leak. Maybe "leakable". That is also a bit confusing because there other objects than these passed to the abstract function too. They don't "leak" because they were already in the outside world.

Reply:
+1 for "leaked objects" terminology. Functions are "pure", but I've never heard of "(im)pure" being applied objects.

"They are not necessarily going to get returned from the pure function so they won't always leak that way." Being returned is not a required, just being accessible by a function is sufficient to be considered as "leaked".

As far as I can see "leakable" is really "leaked".

Reply:
+1 for "leaked objects" terminology. Being returns is not a requirement for "leaked".
"That is also a bit confusing because there other objects than these passed to the abstract function too." Indeed. How about "createdLeakedObjects"?

Reply:
I'm hesitant to call them "leaked" since they won't all have the "hasLeaked" flag.

`objectsTrackedForLeaks`?